### PR TITLE
Ginkgo Logs improvements.

### PR DESCRIPTION
**Pull request by eloycoto**
This PR changes a few things:

- The By message will accept a variadic argument and  no need to use `fmt.Sprintf()`
- All GinkgoWriter data will dump to test-output.log
- New helpers function GinkgoPrint
- In case of a failure, the AfterFailed commands will report in the Jenkins too (To give context to developers)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

File: test/config/logs.go [hunk 1]
```diff
@@ -22,6 +22,9 @@ var (
 	// TestLogFileName is the file name to dump `TestLogWriter` content when
 	// test finish
 	TestLogFileName = "test-output.log"
+
+	// JenkinsLogFilename is the file name where the Jenkins output is saved
+	JenkinsLogFilename = "Jenkin-output.log"
```
Comment (ianvernon) [inline test/config/logs.go]: Jenkin --> jenkins

Comment (ianvernon) [reply test/config/logs.go]: Can you be more descriptive than 'Jenkins output'? What exactly constitutes 'Jenkins output'?

**Review by ianvernon — APPROVED** [review]
A few small changes needed, nothing blocking. No need for re-review.

File: test/config/logs.go (hunk 1 above)
Comment (eloycoto) [reply test/config/logs.go]: Hi, it's a leftover. I'll delete this one, sorry!

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
Hit DNS issue: 
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/3202/testReport/junit/k8s-1/10/K8sValidatedPolicyTest_GuestBook_Examples_checks_policy_example/

**Comment by eloycoto** [comment]
test-me-please

**Pull request merged**